armsrc/hitagS.c

```diff
@@ -387,21 +402,11 @@ static void hts_stop_clock(void) {
  */
 static int check_select(const uint8_t *rx, uint32_t uid) {
 
-    unsigned char resp[48];
-    uint32_t ans = 0x0;
-
-    for (int i = 0; i < 48; i++) {
-        resp[i] = (rx[i / 8] >> (7 - (i % 8))) & 0x1;
-    }
-
-    for (int i = 0; i < 32; i++) {
-        ans += resp[5 + i] << (31 - i);
-    }
-
     // global var?
-    temp_uid = ans;
+    concatbits((uint8_t *)&reader_selected_uid, 0, rx, 5, 32);
```

Comment:
I am not entirely sure what is going on here, but when using the simulator, this instruction sets the variable `m` to 0. So maybe some out-of-bounds write or something?

Reply:
Appreciate the heads-up, concatbits erased out-of-bounds, i'll fix in a new PR